File: Divyansh Singh/Stack/Implement queue using stack.cpp
class MyQueue {
public:
    stack<int>st1;
    stack<int>st2;
    int front;
    MyQueue() {
        
    }
    
    void push(int x) {
        if(st1.empty()){
            front=x;
        }
        st1.push(x);
    }
    
    int pop() {
        if(st2.empty()){
            while(!st1.empty()){
                st2.push(st1.top());
                st1.pop();
            }
            int a=st2.top();
            st2.pop();
            return a;
        }
        int a=st2.top();
        st2.pop();
        return a;
    }
    
    int peek() {
        return st2.empty() ? front : st2.top();
    }
    
    bool empty() {
        if(st1.empty() && st2.empty()){
            return true;
        }
        return false;
    }
};

File: Bhumika Agrawal/Stack/Implement Queue using Stack.java
class MyQueue {
    Stack<Integer>stack1;
    Stack<Integer>stack2;
    int front;
    public MyQueue() {
        stack1=new Stack<>();
        stack2=new Stack<>();
        front=-1;
        
    }
    
    public void push(int x) {
        if(stack1.isEmpty()){
            front=x;
        }
        stack1.push(x);
        
    }
    
    public int pop() {
        if(stack2.isEmpty()){
            while(!stack1.isEmpty()){
                stack2.push(stack1.pop());
            }
        }
        return stack2.pop();
        
    }
    
    public int peek() {
        return stack2.isEmpty()?front:stack2.peek();
    }
    
    public boolean empty() {
        return stack1.isEmpty() && stack2.isEmpty();
        
    }
}
//Time=O(1)
//space=O(n)

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue obj = new MyQueue();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.peek();
 * boolean param_4 = obj.empty();
 */
